Zend/zend_language_scanner.l

```diff
@@ -1551,7 +1551,12 @@ NEWLINE ("\r"|"\n"|"\r\n")
 	RETURN_TOKEN_WITH_IDENT(T_TRAIT);
 }
 
-<ST_IN_SCRIPTING>"enum" {
+/*
+ * The enum keyword must be followed by whitespace and another identifier.
+ * This avoid the BC break of using enum in classes, namespaces, functions and constants.
+ */
+<ST_IN_SCRIPTING>"enum"{WHITESPACE}[a-zA-Z0-9_\x80-\xff] {
```

Comment:
Shouldn't include `0-9` if we're being pedantic, as this is the first char of the label.

Reply:
Right, done.